contract/src/lib.rs

```diff
@@ -251,108 +242,97 @@ mod tests {
-            assert_eq!(contract.balanceOf(samAddress.clone()), 10000.into());
-            assert_eq!(contract.balanceOf(owner.clone()), 30000.into());
-
-            // The limit has reached. No more coins should be available to spend for the spender
-            assert_eq!(contract.transferFrom(owner.clone(), samAddress.clone(), 1.into()), false);
-            assert_eq!(contract.balanceOf(samAddress.clone()), 10000.into());
-            assert_eq!(contract.balanceOf(owner.clone()), 30000.into());
-            assert_eq!(get_external::<ExternalInstance>().logs().len(), 2, "Two events should be created");
-        }
-    );
-
-    test_with_external!(
-        ExternalBuilder::new().build(),
-        spender_should_not_be_able_to_spend_if_owner_has_no_coins {
-            let mut contract = TokenContractInstance{};
-            let owner: Address = Address::new();
-            let spender: Address = "0xdb6fd484cfa46eeeb73c71edee823e4812f9e2e1".into();
-            let samAddress: Address = "0xea674fdde714fd979de3edf0f56aa9716b898ec8".into();
-            contract.constructor(70000.into());
-            contract.transfer(samAddress, 30000.into());
-            contract.approve(spender, 40000.into());
-
-            // Build different external with sender = spender
-            let spenderExternal = ExternalBuilder::from(get_external::<ExternalInstance>())
-                .sender(spender)
-                .build();
-            set_external(Box::new(spenderExternal));
-
-            // Despite of the allowance, can't transfer because the owner is out of tokens
-            assert_eq!(contract.transferFrom(owner.clone(), samAddress.clone(), 40001.into()), false);
-            assert_eq!(contract.balanceOf(samAddress.clone()), 30000.into());
-            assert_eq!(contract.balanceOf(owner.clone()), 40000.into());
-            assert_eq!(get_external::<ExternalInstance>().logs().len(), 0, "Should be no events created");
-        }
-    );
+    #[test]
+    fn approve_should_approve() {
+        ext_reset(|e| e);
+        let mut contract = TokenContractInstance{};
+        let spender: Address = "0xdb6fd484cfa46eeeb73c71edee823e4812f9e2e1".into();
+        contract.constructor(40000.into());
+        contract.approve(spender, 40000.into());
+        assert_eq!(ext_get().logs().len(), 1, "Should be 1 event logged");
+        assert_eq!(ext_get().logs()[0].topics.as_ref(), &[
+            "0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925".into(), // hash of the event name
+            "0x0000000000000000000000000000000000000000000000000000000000000000".into(), // sender (owner) address
+            "0x000000000000000000000000db6fd484cfa46eeeb73c71edee823e4812f9e2e1".into()]); // spender address
+        assert_eq!(contract.allowance(::pwasm_ethereum::sender(), spender.clone()), 40000.into());
+    }
+
+    #[test]
+    fn spender_should_be_able_to_spend_if_allowed() {
+        ext_reset(|e| e);
```

Comment:
wut this construction mean?
just reset to some clear state?

Reply:
exactly. In this case we just need a fresh External.